test/commands/crashes/upload-mappings/upload-mappings-test.ts

```diff
@@ -0,0 +1,244 @@
+import { expect, use } from "chai";
+import * as Fs from "fs";
+import * as _ from "lodash";
+import * as Nock from "nock";
+import * as Path from "path";
+import * as Sinon from "sinon";
+import * as Temp from "temp";
+import * as ChaiAsPromised from "chai-as-promised";
+import * as MockRequire from "mock-require";
+
+use(ChaiAsPromised);
+
+// Mocking AzureBlobUploadHelper
+import AzureBlobUploadHelperMock from "../lib/azure-blob-uploader-helper-mock";
+MockRequire("../../../../src/commands/crashes/lib/azure-blob-upload-helper", {
+  default: AzureBlobUploadHelperMock
+});
+import UploadMappingsCommand from "../../../../src/commands/crashes/upload-mappings";
+MockRequire.stopAll();
+
+import { CommandArgs, CommandResult } from "../../../../src/util/commandline";
+
+Temp.track();
+
+describe("upload-mappings command", () => {
+  const fakeAppOwner = "fakeAppOwner";
+  const fakeAppName = "fakeAppName";
+  const fakeAppIdentifier = `${fakeAppOwner}/${fakeAppName}`;
+  const fakeToken = "c1o3d3e7";
+  const fakeSymbolUploadingId = "fakeSymbolUploadingId";
+  const fakeUploadUrl = "/upload/here";
+  /* tslint:disable-next-line:no-http-string */
+  const fakeHost = "http://localhost:1700";
+  const fakeFullUploadUrl = fakeHost + fakeUploadUrl;
+
+  const mappingsFileName = "mapping.txt";
+
+  const mappingsFileContent = "Mappings";
+
+  let tmpFolderPath: string;
+
+  let postSymbolSpy: Sinon.SinonSpy;
+  let patchSymbolSpy: Sinon.SinonSpy;
+  let abortSymbolUploadSpy: Sinon.SinonSpy;
+
+  let expectedRequestsScope: Nock.Scope;
+  let skippedRequestsScope: Nock.Scope;
+
+  before(() => {
+    Nock.disableNetConnect();
+  });
+
+  beforeEach(() => {
+    tmpFolderPath = Temp.mkdirSync("uploadMappingsTest");
+    postSymbolSpy = Sinon.spy();
+    patchSymbolSpy = Sinon.spy();
+    abortSymbolUploadSpy = Sinon.spy();
+  });
+
+  describe("input validation", () => {
```

Comment:
Some additional test ideas:
- If we only want to support `mapping.txt` as a valid filename, we could test for that here
- If `version` (I believe) _must_ be a number, we should check that we can cast the string input to a number

Reply:
Yes on the account that `version code` needs to be a positive (non-zero) integer, will add a validation here. Typically this should get caught by the upload pipeline, but it is a restriction imposed by the Android SDK, not App Center, so we can require it in here, too.